### PR TITLE
166 move background initialization to the main correctbackgroundaccurate pipeline step

### PR DESCRIPTION
**Pull request by nepstad**

### CODE REVIEW

File: pyopia/pipeline.py [hunk 1]
```diff
@@ -195,12 +193,12 @@ class Data(TypedDict):
...
     bgstack: float
     '''List of images making up the background (either static or moving)
-    Obtained from :class:`pyopia.background.CreateBackground`
+    Obtained from :class:`pyopia.background.CorrectBackgroundAccurate`
     '''
     imbg: float
     '''Background image that can be used to correct :attr:`pyopia.pipeline.Data.imraw`
     and calcaulte :attr:`pyopia.pipeline.Data.imc`
-    Obtained from :class:`pyopia.background.CreateBackground`
+    Obtained from :class:`pyopia.background.CorrectBackgroundAccurater`
```
Comment (emlynjdavies) [inline pyopia/pipeline.py:201]: CorrectBackgroundAccurater should be CorrectBackgroundAccurate

**Review by emlynjdavies — CHANGES_REQUESTED** [review]
I get this error in pipline-holo notebook:

```
IndexError                                Traceback (most recent call last)


     plt.style.use('dark_background')
     zval = np.arange(pipeline_steps['initial']['minZ'],
     pipeline_steps['initial']['maxZ'],
     pipeline_steps['initial']['stepZ'])
     plt.hist(zval[xstats.ifocus-1],len(zval))
     plt.xlim(zval[0],zval[-1])
     plt.xlabel('Z [mm]')

IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

**Comment by emlynjdavies** [comment]
> I get this error in pipline-holo notebook:
> 
> ```
> IndexError                                Traceback (most recent call last)
> 
> 
>      plt.style.use('dark_background')
>      zval = np.arange(pipeline_steps['initial']['minZ'],
>      pipeline_steps['initial']['maxZ'],
>      pipeline_steps['initial']['stepZ'])
>      plt.hist(zval[xstats.ifocus-1],len(zval))
>      plt.xlim(zval[0],zval[-1])
>      plt.xlabel('Z [mm]')
> 
> IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
> ```

NOTE: due to xstats.ifocus not being int

**Comment by emlynjdavies** [comment]
This seems to work in the holo-pipeline notebook now (have not checked anything else yet).

However, I noticed that the ROIs in the montage are very dark and strange-looking. I think this happened first from version 1.1.3. Is the 255 factor that was added there, causing problems? What are your thoughts, @animmosmith ?

**Review by emlynjdavies — CHANGES_REQUESTED** [review]
Previously we could run the pipeline on a single image, like this:

MyPipeline = pyopia.pipeline.Pipeline(settings) # this line would create an initial background
stats = MyPipeline.run(filename) # this would process a single image using the background created in the line above

Now, it seems that always the first image we process will be zeroed out. Is there a way to maintain the above functionality?

**Comment by animmosmith** [comment]
> This seems to work in the holo-pipeline notebook now (have not checked anything else yet).
> 
> However, I noticed that the ROIs in the montage are very dark and strange-looking. I think this happened first from version 1.1.3. Is the 255 factor that was added there, causing problems? (this is a separate issue relating to PR #155) What are your thoughts, @animmosmith ?

Yes - there is an error here - I don't think "rescale_stack" is ever applied but I need to track that through

**Review by animmosmith — DISMISSED** [review]

**Comment by emlynjdavies** [comment]
> > This seems to work in the holo-pipeline notebook now (have not checked anything else yet).
> > However, I noticed that the ROIs in the montage are very dark and strange-looking. I think this happened first from version 1.1.3. Is the 255 factor that was added there, causing problems? (this is a separate issue relating to PR #155) What are your thoughts, @animmosmith ?
> 
> Yes - there is an error here - I don't think "rescale_stack" is ever applied but I need to track that through. I'll raise that as a separate Issue.

Thanks @animmosmith We are working on a fix here: https://github.com/SINTEF/pyopia/pull/174
and I think we need to take a chat about expected output value ranges from the reconstruction

**Review by emlynjdavies — CHANGES_REQUESTED** [review]
Suggest this will be part of version 1.2 as it introduced breaking changes. Perhaps this, and #91 should be combined first, and then merged to main as version 1.2.0?

**Review by emlynjdavies — APPROVED** [review]

**Comment by emlynjdavies** [comment]
fixes #164

**Pull request merged**